legion/engine/rendering/data/shader.hpp

```diff
@@ -101,7 +101,11 @@ namespace legion::rendering
                 tex = invalid_texture_handle.get_texture();
 
             glActiveTexture(GL_TEXTURE0 + m_textureUnit);
-            glBindTexture(GL_TEXTURE_2D, tex.textureId);
+            if (tex.type == texture_type::two_dimensional)
+                glBindTexture(GL_TEXTURE_2D, tex.textureId);
+            else if (tex.type == texture_type::array_2D)
+                glBindTexture(GL_TEXTURE_2D_ARRAY, tex.textureId);
```

Comment:
i mean honestly, the value of `texture_type::two_dimensional` is `GL_TEXTURE_2D` and such. so couldn't you just do `glBindTexture(static_cast<GLenum>(tex.type)`?